deployment/ansible/roles/nginx-certbot-deploy/tasks/main.yml

```diff
@@ -44,3 +44,66 @@
         - /judgels/sites:/etc/nginx/conf.d
       env:
         CERTBOT_EMAIL: "{{ letsencrypt_email }}"
+
+  - name: Allow Nginx HTTP traffic through the firewall
+    ufw:
+      rule: allow
+      to_port: 80
+      proto: tcp
+      comment: Judgels Nginx HTTP
+
+  - name: Allow Nginx HTTPS traffic through the firewall
+    ufw:
+      rule: allow
+      to_port: 443
+      proto: tcp
+      comment: Judgels Nginx HTTPS
+
+  - name: Delete Jophiel traffic allow rule
```

Comment:
Why do we have this? Then we will kill all ufw rules for Judgels whenever we redeploy nginx-certbot?

Reply:
Each role for Judgels services adds rules than open ports used by each service. When we use nginx-certbot, we no longer need those allow rules.

Reply:
Yes I know, but why? Then whenever we redeploy nginx-certbot (e.g. when previous cert check fails), then all Judgels services UFW rule will be reset? I mean then we have to readd UFW rule for Judgels services whenever we redeploy this.

Reply:
The allow rules are only needed if we don't use nginx-certbot. If we use nginx-certbot, we won't need the allow rules for the other Judgels services since we should only be able to access those services through Nginx.

Reply:
Update: reverted to prevent nginx-certbot playbook from disabling existing allow rules for other services, due to the confusion that may happen e.g. if we redeploy other Judgels services after deploying nginx-certbot (closed ports may reopen) or if we decide to undo deploying nginx-certbot (need to reopen other services' ports manually)